compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -258,7 +258,18 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 Ok(ty::Visibility::Restricted(DefId::local(CRATE_DEF_INDEX)))
             }
             ast::VisibilityKind::Inherited => {
-                Ok(ty::Visibility::Restricted(parent_scope.module.normal_ancestor_id))
+                if self.parent_scope.module.kind.is_enum() {
+                    // Anything in an enum has the visibility of the enum.
+                    Ok(*self
+                        .r
+                        .visibilities
+                        .get(&self.parent_scope.module.normal_ancestor_id.expect_local())
```

Comment:
```suggestion
                        visibilities[&self.parent_scope.module.def_id().unwrap().expect_local()]
```
By using `normal_ancestor_id` you don't use visibility of enum, but rather visibility of its parent `mod` item.

Reply:
Ah, okay! I wasn't sure what it did.

Related to this, what does `nearest_item_scope()` do? (See my comment at https://github.com/rust-lang/rust/pull/79956#discussion_r541495180.)